project_templates/roundtable_aiohttp_bot/{{cookiecutter.name}}/manifests/base/deployment.yaml

```diff
@@ -25,3 +26,13 @@ spec:
           envFrom:
             - configMapRef:
                 name: {{ cookiecutter.package_name }}
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - all
+            readOnlyRootFilesystem: true
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 1000
```

Comment:
Quick question: does this need to match up with the user set in the Dockerfile for a given container; or does it just work (or isn't related?).

Reply:
My understanding is that `runAsUser` overrides the `USER` directive in the `Dockerfile` and instead always starts the container with the specified user. However, all of our containers use `adduser` to create a user, and `adduser` will by default create a user with UID and GID 1000, so the end result is the same. The difference is that this removes control of that from the Docker image and changes users earlier in the launch process and, more important for our purposes, it unblocks eventually setting a `PodSecurityPolicy` on our clusters. If a `PodSecurityPolicy` is set, Kubernetes will refuse to launch pods that do not specify a `runAsUser`.